### PR TITLE
Update TitaNet-L and MSDD models 

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

* Update hydra config on published NGC titanet-model and release as v1 version
* Improve speaker label tests with latest hydra config
* Add a feature to write model labels `labels.txt` to exp directory 
* Update batch inference function to return gt labels and trained model labels (if available) 

**Collection**: ASR

# Changelog 
- Update label models to include latest v1 version of titanet-large
- Added titanet-small config file (model will be published soon)
- Now instead of returning index of gt labels from AudioSpeechLabelDataset, return the speaker label names directly as seen from manifest file

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [x] Documentation

### CODE REVIEW

**Comment by nithinraok** [comment]
test `test_pretrained_titanet_embeddings` would be passed after v1 version of model gets published to NGC [ON GOING]

File: nemo/collections/asr/models/label_models.py [hunk 1]
```diff
@@ -124,21 +122,6 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
             weight = [sum(self.labels_occurrence) / (len(self.labels_occurrence) * i) for i in self.labels_occurrence]
 
         if 'loss' in cfg:
-            # To support older version checkpoints
```
Comment (fayejf) [inline nemo/collections/asr/models/label_models.py:127]: how about older checkpoints such as speakernet?

File: nemo/collections/asr/models/label_models.py [hunk 2]
```diff
@@ -546,21 +528,20 @@ def batch_inference(self, manifest_filepath, batch_size=32, sample_rate=16000, d
             embs: embeddings of files provided in manifest file
             logits: logits of final layer of EncDecSpeakerLabel Model
             gt_labels: labels from manifest file (needed for speaker enrollment and testing)
-            mapped_labels: Classification labels sorted in the order that they are mapped by the trained model
+            trained_labels: Classification labels sorted in the order that they are mapped by the trained model
```
Comment (fayejf) [inline nemo/collections/asr/models/label_models.py:531]: how about the groundtruth label does not belong to trained_label?  What whould be gt_labels in that case? 
If gt_label does not present and the model hasn't seen the class the audio, it will output most likely pred?

File: nemo/collections/asr/models/label_models.py (hunk 1 above)
Comment (nithinraok) [reply nemo/collections/asr/models/label_models.py:127]: Latest versions are being published on NGC. Urls would be updated accordingly

File: nemo/collections/asr/models/label_models.py (hunk 2 above)
Comment (nithinraok) [reply nemo/collections/asr/models/label_models.py:531]: gt_lables doesn;t depend on trained labels now, they just return what it gets from manifest.

File: nemo/collections/asr/models/label_models.py (hunk 1 above)
Comment (fayejf) [reply nemo/collections/asr/models/label_models.py:127]: Then let's merge and update the link until the latest version is published? Or it will break things.

File: nemo/collections/asr/models/label_models.py (hunk 2 above)
Comment (fayejf) [reply nemo/collections/asr/models/label_models.py:531]: then could you add a test to ensure these conditions, please?

File: nemo/collections/asr/models/label_models.py (hunk 1 above)
Comment (nithinraok) [reply nemo/collections/asr/models/label_models.py:127]: Added links for speakernet and ecapa models as well. All models are updated now

File: nemo/collections/asr/models/label_models.py (hunk 2 above)
Comment (fayejf) [reply nemo/collections/asr/models/label_models.py:531]: No, we didn't have test for groundtruth label does not belong to trained_label. Could you add this?
